packages/ember-application/lib/system/application.js

```diff
@@ -253,31 +254,44 @@ var Application = Namespace.extend(DeferredMixin, {
   */
   customEvents: null,
 
+  /**
+    Whether the application should automatically start routing and render
+    templates to the `rootElement` on DOM ready. While default by true,
+    other environments such as FastBoot or a testing harness can set this
+    property to `false` and control the precise timing and behavior of the boot
+    process.
+
+    @property autoboot
+    @type Boolean
+    @default true
+    @private
+  */
+  autoboot: true,
+
   init: function() {
     this._super.apply(this, arguments);
 
-    // Start off the number of deferrals at 1. This will be
-    // decremented by the Application's own `initialize` method.
-    this._readinessDeferrals = 1;
-
     if (!this.$) {
       this.$ = jQuery;
     }
 
-    // Create subclass of Ember.Router for this Application instance.
-    // This is to ensure that someone reopening `App.Router` does not
-    // tamper with the default `Ember.Router`.
-    // 2.0TODO: Can we move this into a globals-mode-only library?
-    this.Router = Router.extend();
     this.buildRegistry();
 
-    // TODO:(tomdale+wycats) Move to session creation phase
-    this.buildInstance();
-
     registerLibraries();
     logLibraryVersions();
 
-    this.scheduleInitialize();
+    // Start off the number of deferrals at 1. This will be
+    // decremented by the Application's own `initialize` method.
+    this._readinessDeferrals = 1;
+
+    if (!Ember.FEATURES.isEnabled('ember-application-visit') || this.autoboot) {
```

Comment:
The feature flag system cannot handle this properly.

It **must** be precisely:

```javascript
if (Ember.FEATURES.isEnabled('ember-application-visit')) {
  /* stuff when enabled */
} else {
  /* stuff when disabled */
}
```